src/editfns.c

```diff
@@ -1098,38 +1098,6 @@ At the beginning of the buffer or accessible region, return 0.  */)
   return temp;
 }
 
-DEFUN ("char-after", Fchar_after, Schar_after, 0, 1, 0,
-       doc: /* Return character in current buffer at position POS.
-POS is an integer or a marker and defaults to point.
-If POS is out of range, the value is nil.  */)
-  (Lisp_Object pos)
-{
-  register ptrdiff_t pos_byte;
-
-  if (NILP (pos))
-    {
-      pos_byte = PT_BYTE;
-      XSETFASTINT (pos, PT);
-    }
-
-  if (MARKERP (pos))
-    {
-      pos_byte = marker_byte_position (pos);
-      if (pos_byte < BEGV_BYTE || pos_byte >= ZV_BYTE)
-	return Qnil;
-    }
```

Comment:
This looks to me like a bug. `BEGV_BYTE` and `ZV_BYTE` belong to the current buffer instead of the buffer the marker points to.

Reply:
Well, the docstring says that it uses the current buffer. So if the marker points to a different buffer the offset is still used, for the current buffer.

Not very clear, I agree, but appears nevertheless intended.

Reply:
I can't imagine anybody is expecting this behaviour. Why use a marker if you don't use its location ? I will ask upstream what they think about it.